### PR TITLE
Etv 7283 fix optional inputs

### PR DESCRIPTION
**Pull request by hututudor**
https://etvas.atlassian.net/browse/ETV-7283

### CODE REVIEW

File: src/Autocomplete/Autocomplete.jsx [hunk 1]
```diff
@@ -193,7 +196,8 @@ Autocomplete.defaultProps = {
     value ? (Array.isArray(value) ? value.includes(v) : value === v) : false,
   onChange: () => console.warn('Autocomplete.onChange should be a function'),
   placeholder: 'Please select an option',
-  searchMaxResults: 30
+  searchMaxResults: 30,
+  optionalText: 'Optional'
```
Comment (ohnu) [inline src/Autocomplete/Autocomplete.jsx:200]: how does this work with the localization requirements ?

File: src/TextArea/TextArea.jsx [hunk 2]
```diff
@@ -53,13 +55,27 @@ export const TextArea = forwardRef(
...
+                ml={1}
+                as='label'
+                htmlFor={id}
+                variant='base12Bold'
+                color='baseGray'
+                width='fit-content'>
+                - {optionalText}
+              </Typography>
+            )}
+          </Box>
```
Comment (calinbogdan) [inline src/TextArea/TextArea.jsx:78]: Please use extracted Label component here

File: src/Autocomplete/Autocomplete.jsx (hunk 1 above)
Comment (hututudor) [reply src/Autocomplete/Autocomplete.jsx:200]: In en/de the word is the same, but I intend to provide this prop when using the component.

File: src/PhoneNumberInput/PhoneNumberInput.jsx [hunk 3]
```diff
@@ -209,13 +211,27 @@ const PhoneNumberInput = forwardRef((props, ref) => {
...
+              ml={1}
+              as='label'
+              htmlFor={id}
+              variant='base12Bold'
+              color='baseGray'
+              width='fit-content'>
+              - {optionalText}
+            </Typography>
+          )}
+        </Box>
```
Comment (calinbogdan) [inline src/PhoneNumberInput/PhoneNumberInput.jsx:234]: Use the extracted label component here

File: src/SubdomainInput/SubdomainInput.jsx [hunk 4]
```diff
@@ -152,13 +154,27 @@ export const SubdomainInput = forwardRef(
...
+                ml={1}
+                as='label'
+                htmlFor={id}
+                variant='base12Bold'
+                color='baseGray'
+                width='fit-content'>
+                - {optionalText}
+              </Typography>
+            )}
+          </Box>
```
Comment (calinbogdan) [inline src/SubdomainInput/SubdomainInput.jsx:177]: Use extracted label component

File: src/Form/TextAreaField.jsx [hunk 5]
```diff
@@ -29,9 +29,11 @@ const hasValidation = props => props.validate || props.required
 TextAreaField.propTypes = {
   ...fieldShape,
   placeholder: PropTypes.node,
-  required: PropTypes.bool
+  required: PropTypes.bool,
+  optionalText: PropTypes.node
```
Comment (calinbogdan) [inline src/Form/TextAreaField.jsx:33]: It seems like this is added in the fieldShape object. Do we still need it?

**Review by ohnu — APPROVED** [review]

File: src/Dropdown/Dropdown.jsx [hunk 6]
```diff
@@ -243,13 +245,27 @@ const Dropdown = ({
...
+              ml={1}
+              as='label'
+              htmlFor={id}
+              variant='base12Bold'
+              color='baseGray'
+              width='fit-content'>
+              - {optionalText}
+            </Typography>
+          )}
+        </Box>
```
Comment (calinbogdan) [inline src/Dropdown/Dropdown.jsx:268]: I think we should extract this into a component:

```
<Label inputId={id} showOptionalText={!required} optionalText={string?}/>
```

We can set the default prop value for this new Label component to 'Optional' and we don't need to set the default prop in every field that uses it

**Pull request merged**